src/endpoint/s3/s3_errors.js

```diff
@@ -512,7 +513,7 @@ S3Error.XAmzContentSHA256Mismatch = Object.freeze({
 });
 S3Error.MalformedPolicy = Object.freeze({
     code: 'MalformedPolicy',
-    message: 'Invalid principal in policy',
+    message: 'Policy was not well-formed or did not validate against the published schema.',
```

Comment:
Why did you change this?
In the attached image, I can see that the message is from the error `MalformedXML` (not `MalformedXML`).

Reply:
Previously, this customized msg was added for `MalformedPolicy` error we are throwing in case of `invalid_schema_params` error, for which we have now created a function `invalid_schema_to_aws_error`. So, I have update the common error msg here. wdyt, do we need to update here?

Reply:
You need to see what AWS does and be compatible with it.

Reply:
as per aws documentation [here](https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html) , we can update the error msg to `"Your policy contains a principal that is not valid."`

Reply:
I think it's a description, not the actual message.

Reply:
![Screenshot from 2025-01-15 21-45-03](https://github.com/user-attachments/assets/643cd4f6-998f-46eb-b03c-d3079441d299)

yes you are correct, removed these changes. Thanks